pkg/webhook/webhook.go

```diff
@@ -621,6 +622,60 @@ func getResourceList(resourceRequests map[string]int64) *corev1.ResourceList {
 	return &resourceList
 }
 
+func createCustomizedPatch(pod corev1.Pod) ([]jsonPatchOperation, error) {
+	var patch jsonPatchOperation
+	var customizedPatch []jsonPatchOperation
+
+	for cik, civ := range customizedInjections {
+		// The customizedInjections will be injected when:
+		// 1. Pod labels contain the patch key defined in customizedInjections, and
+		// 2. The value of patch key in pod labels(not in customizedInjections) is "true"
+		if podValue, exists := pod.ObjectMeta.Labels[cik]; exists && podValue == "true" {
+			err := json.Unmarshal([]byte(civ), &patch)
+			if err != nil {
+				return nil, err
+			}
+			customizedPatch = append(customizedPatch, patch)
+		}
+	}
+	for _, p := range customizedPatch {
```

Comment:
can this validation happen on set of `customizedInjections`  instead of every pod creation?

Reply:
`customizedPatch` contains only patch that a pod requests. in the for loop above, there is check on which labels are requested by pod and only add the requested customized patch in `customizedPatch`.
I assume we cannot pre-predict which labels user may set on pod.

Reply:
it is necessarily a subset of `customizedInjections ` isnt it ?
so validating that each patch provided in `customizedInjections` once when setting it will not require to check it per Pod.
it seems to me more efficient.

This does not affect the correctness of the code, and is a suggestion on how to make it a bit more efficient as less operations will be performed when processing a Pod.

Reply:
Moved this down to `SetCustomizedInjection()` so that it only be executed once.